load-test-framework/src/main/java/com/google/pubsub/flic/controllers/GCEController.java

```diff
@@ -0,0 +1,331 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+package com.google.pubsub.flic.controllers;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.InputStreamContent;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.*;
+import com.google.api.services.storage.Storage;
+import com.google.api.services.storage.model.Bucket;
+import com.google.common.base.Preconditions;
+import com.google.pubsub.flic.controllers.Client.ClientType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+class GCEController extends Controller {
+  private static final Logger log = LoggerFactory.getLogger(GCEController.class.getName());
+  private final Storage storage;
+  private final Compute compute;
+  private final Executor executor;
+  private List<ClientType> types;
+  private boolean shutdown;
+  private final String machineType = "n1-standard-4"; // quad core machines
+  private final String sourceFamily = "projects/debian-cloud/global/images/family/debian-8"; // latest Debian 8
+  private final String projectName;
+  private final String zone = "us-central1-a";
```

Comment:
This is just a placeholder since as of right now this is just a single zone. Eventually I'm going to make the constructor argument
Map<Zone, Map<ClientType, Integer>> and then we can have arbitrary numbers of client types in different zones that all work. My plan is to do that in another PR (this one seemed long enough to me already, but I can do it in this one if you feel strongly about it).